fix(systemd): explicitly install some libs that will not be statically included

aafeijoo-suse opened the pull request:
Some required libraries that used to be statically included are in the process to be opened via `dlopen()`.

## Checklist
- [ ] I have tested it locally
- [ ] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

Closes #2642

ddiss reviewed (approved):
I think it'd help to have some of the corresponding systemd commit ids present in the commit message (once merged there).
Anyhow, this looks good to me. I assume you're waiting for the systemd changes to go through before dropping the Draft flag?

ddiss commented on modules.d/01systemd-journald/module-setup.sh:
```diff
@@ -53,9 +53,10 @@ install() {
-        {"tls/$_arch/",tls/,"$_arch/",}"libzstd.so.*" \
-        {"tls/$_arch/",tls/,"$_arch/",}"liblzma.so.*"
+        {"tls/$_arch/",tls/,"$_arch/",}"liblzma.so.*" \
+        {"tls/$_arch/",tls/,"$_arch/",}"libzstd.so.*"
```
Could `$hostonly` images selectively drop some of these modules, depending on `systemd-journal` configuration?

aafeijoo-suse replied:
I know the default compression is configured at build time, but it can also be set via `SYSTEMD_JOURNAL_COMPRESS` env var... I'll take a look

aafeijoo-suse commented:
> I think it'd help to have some of the corresponding systemd commit ids present in the commit message (once merged there). Anyhow, this looks good to me. I assume you're waiting for the systemd changes to go through before dropping the Draft flag?

Thanks, that's what I was thinking. I'll remove the draft mark and amend the commit message including those external references if they are merged.

aafeijoo-suse committed:
fix(systemd): explicitly install some libs that will not be statically included

Some required libraries that used to be statically included are in the process
to be opened via `dlopen()`.

References:
- https://github.com/systemd/systemd/pull/31131
- https://github.com/systemd/systemd/pull/31550
- https://github.com/systemd/systemd/pull/32019

Closes #2642

pvalena commented:
LGTM.
